src/solver/optim-model-filler/ComponentFiller.cpp

```diff
@@ -232,13 +228,15 @@ void ComponentFiller::addTimeDependentConstraints(
         {
             if (IsThisVariableTimeDependent(var_id))
             {
-                auto* variable = pb.getVariable(component_.Id() + "." + var_id + '_'
-                                                + std::to_string(constraint_count));
+                // TODO scenario = 0
+                auto* variable = variableDict(component_.Id(), var_id, constraint_count, 0);
+
                 ct->setCoefficient(variable, coef);
             }
             else
             {
-                auto* variable = pb.getVariable(component_.Id() + "." + var_id);
+                auto* variable = variableDict(component_.Id(), var_id, 0, 0);
```

Comment:
why?

Reply:
Because this variable is time & scenario independent

Reply:
Then it should not be tied to time and scenario 0